# The objective of this document is to create an easy to use script for estimating sample size of an RNA-sequencing experiment whiel utilizing 
# the 'ncell' package created by the Jackson Laboratory.

# In order to utilize this package, one should have a reference data set similar to the subpopulation of desired estimation to propperly asses the arguments


```{r load libraries and package}
library('devtools')
install_github('TheJacksonLaboratory/ncells')
library('ncells')
```


# Arguments in 'ncells'
#function (m1, pi1, foldchange, dropout, p = 20000, n = seq(100, 
#    1000, by = 100), mu = 1, sigma = 5, type1 = 0.05, dfactor, 
#    seed = 123, B = 1000, ncore = 1) 

#Example:
# ncells(m1=1.6, pi1=5, foldchange=4, dropout=.6, p=20000, mu=1, sigma=5)

```{r define argument; followed by range of argument}
m = 0.1                    # fraction of marker genes;  0.1 <= m <= 10
pi = 5                     # Percentage of desired subpopulation of cells; 5 <= pi <= 50
fc =  1                    # Expression fold change of desired subpopulation; 1 <= fc <= 32
do =  0.6                  # dropout rate; 0.6 <= do <= 0.999
unique_gene_count = 10000  # Total number of unique genes in simulation; 10000 <= unique_gene_count
number_cells = 10000       # Vector of total number of cells; 50 <= min(number_cells), max(number_cells) <= 1000000
mean_expression =  1       # (default = 1)
standard_deviation=  5     # Standard deviation of mean gene expressions (default = 5)
alpha = 0.05               # type 1 error rate, default = 0.05
#dfactor                   # Option, Cutoff values to declare separation, depends on type 1 error rate
seed_number = 123          # random seed, default 123
replicates =  1000         # Number of replicates to estiamte the probability of success separation, default 1000, 100 <= replicates
cores = 1                  # number of cores in the simulation, default = 1
```


# This may take some time to run
```{r, put it together}

ncells(m1 = m, pi1=pi, foldchange = fc, dropout = do, p = unique_gene_count, n = number_cells, mu = mean_expression, sigma = standard_deviation, type1 = alpha, dfactor, seed = seed_number, B = replicates, ncore = cores)

```

